### ISSUE TITLE
Challenge task parameter values are mixed up when reconfiguring

### ISSUE THREAD

**Issue by MichaelRoeder** [hobbit-project/platform#30]
## Problem
1. The user creates a challenge
2. The user creates a challenge task (and stores it)
3. The user opens the challenge task again and it is possible that the values of parameters are in wrong positions.

The screenshot shows an example of a challenge task, which has been created with the default values
* window size = 10
* seed = 123
* messages = 63
* transitions = 5
* interval = 10
* output = 0
* max clustering = 50
* prob. = 0.005

After reopening the challenge task, the values are placed in the wrong positions.

![challenge_task_value_mixture](https://cloud.githubusercontent.com/assets/6122307/24906765/70c91626-1eb9-11e7-86e6-de18e8652876.png)

**Comment by MichaelRoeder** [comment]
I could figure out that the correct data seems to be transferred from the backend. As it can be seen in the following snippet, the window size of the example above is sent with the value `10` instead of the 0 that is shown in the GUI (see the screenshot above).
```json
{
	"type": "challengeBean",
	"name": "DEBS test challenge",
	...
	"tasks": [{
			"description": "A test task for DEBS",
			"id": "http://w3id.org/hobbit/challenges#08875c20-ee40-419a-8c66-467162535673_task1",
			"name": "DEBS test task 1",
			"benchmark": {
				...
				"configurationParams": [{
						"description": "Window size",
						"id": "http://www.debs2017.org/gc/windowSize",
						"name": "Window size",
						"datatype": "xsd:unsignedInt",
						"defaultValue": "10",
						"feature": true,
						"range": "http://www.w3.org/2001/XMLSchema#unsignedInt",
						"required": false
					}, ...
			},
			"configurationParams": [{
					"description": "Window size",
					"id": "http://www.debs2017.org/gc/windowSize",
					"name": "Window size",
					"datatype": "xsd:unsignedInt",
					"range": "http://www.w3.org/2001/XMLSchema#unsignedInt",
					"value": "10"
				},
                ...
			]
		}
	]
}
```
Maybe the problem is caused by the different order of the parameters in the two `configurationParams` lists?

@romankorf Please have a look at this issue and find the complete json attached.
[challenge_task.txt](https://github.com/hobbit-project/platform/files/913026/challenge_task.txt)

**Comment by romankorf** [comment]
Hi Michael,

Committed unter den branch "hotfix-disableExpResubmission".

https://github.com/hobbit-project/platform/tree/hotfix-disableExpResubmission

Roman



Von:    Michael Röder <notifications@github.com>
An:     hobbit-project/platform <platform@noreply.github.com>
Kopie:  Roman Korf <r.korf@usu.de>, Mention <mention@noreply.github.com>
Datum:  11.04.2017 13:38
Betreff:        Re: [hobbit-project/platform] Challenge task parameter 
values are mixed up when reconfiguring (#30)



Assigned #30 to @romankorf.
?

**Comment by MichaelRoeder** [comment]
I checked out the changes and build the gui with the following steps:
```
~/workspace/platform/hobbit-gui/gui-client$ npm run build

~/workspace/platform/hobbit-gui/gui-serverbackend$ mvn clean package -DskipTests

~/workspace/platform/$ docker-compose build gui
```
After that, I still had the same problem. I repeated the steps to make sure that the build replaced the older version but the problem persists.

@romankorf Can you please make sure that the problem is solved?

**Comment by MichaelRoeder** [comment]
Solved in `v1.0.6`.
